### ISSUE TITLE
Allow passing xticklabels and yticklabels to PolarAxes __init__

### ISSUE THREAD

**Issue by yportier** [matplotlib/matplotlib#15223]
Based on the following question on SO:  

- [Setting xticklabels and yticklabels from subplot_kw in a polar plot](https://stackoverflow.com/questions/57839102/setting-xticklabels-and-yticklabels-from-subplot-kw-in-a-polar-plot)  

It would be nice to allow passing `xticklabels` and `yticklabels` to `PolarAxes` for consistency given that `ax.set_xticklabels()` do work as expected.

**Comment by timhoffm** [comment]
Thanks for the proposal. However, I don't think we should bloat the API of `(Polar)Axes.__init__` with all kind of possible configuration options of the axes. I recommend to use  `ax.set_xticklabels()` on a separate line. Note also that the base class `Axes` itself does not support these parameters either.

**Comment by jklymak** [comment]
While usually I agree w/ not bloating things, I'm not sure I agree that 
```
 _, ax = plt.subplots(subplot_kw=dict(polar=True, xticklabels=[], yticklabels=[]))
```
should behave differently than:
```
 _, ax = plt.subplots(subplot_kw=dict(polar=False, xticklabels=[], yticklabels=[]))
```

<img width="551" alt="ploar" src="https://user-images.githubusercontent.com/1562854/64490649-04031300-d214-11e9-9f63-c73e963d6f05.png">

**Comment by timhoffm** [comment]
@jklymak You are right, I didn't know you could essentially pass all properties as kwargs. Does this not work for polar xticklabels because there's some special-handling code that overrides the parameters?

A similar probelm exists for `xticks` and `xlim`.

**Comment by jklymak** [comment]
Im not sure.  @QuLogic has done a good bit of work on polar and may be able to say.

**Comment by QuLogic** [comment]
This is because `PolarAxes.__init__` calls `self.cla()` after `super().__init__(...)`, which resets everything that would have been set. Presumably, this is to apply the sticky edges and aspect changes, though I'm not totally certain. Just removing it breaks several tests, so sticking a second `self.update` at the end might work.
